book/src/transaction.md

```diff
@@ -0,0 +1,73 @@
+# Anatomy of a Transaction
+
+This chapter documents the binary format of a transaction.
+
+## Transaction Format
+
+A transaction contains a [compact-array](#Compact-Array-Format) of signatures,
+followed by a [message](#Message-Format).  Each item in the signatures array is
+a [digital signature](#Signature-Format) of the given message. The Solana
+runtime verifies that the number of signatures matches the number in the first
+8 bits of the [message header](#Message-Header-Format). It also verifies the
+signature was signed by the private key corresponding to the public key at the
+same index in the message's account addresses vector.
+
+### Signature Format
+
+Each digital signature is in the ed25519 binary format and consumes 64 bytes.
+
+
+## Message Format
+
+A message contains a [header](#Message-Header-Format), followed by a
+compact-array of [account addresses](#Account-Address-Format), followed by a
+recent [blockhash](#Blockhash-Format), followed by a compact-array of
+[instructions](#Instruction-Format).
+
+### Message Header Format
+
+The message header contains three unsigned 8-bit values. The first value is a
+number of required signatures in the containing transaction. The second value
+is the number of read-only account addresses requiring signatures. These will
+be the first addresses in the message's vector of account addresses.  The third
+value in the header is the number of read-only account addresses not requiring
+signatures. These addresses immediately follow the addresses that require
+signatures. The remaining addresses are used to request write access to the
```

Comment:
This section is ambiguous, and I think actually incorrect. The order of account addresses in the account_keys array is:
1. signed read-write keys
2. signed read-only keys
3. unsigned read-write keys
4. unsigned read-only keys.

Here a suggested rewording:
The message header contains three unsigned 8-bit values. The first value is the number of required signatures in the containing transaction. The second value is the number of those corresponding account addresses that are read-only. The addresses that require signatures appear at the beginning of the account address array, with addresses requesting write access first and read-only accounts following.

The third value in the message header is the number of read-only account addresses not requiring signatures. The addresses that do not require signatures follow the addresses that do in the account address array, with read-write accounts first and read-only accounts following.

---
Alternately, perhaps just use the ordered list in some fashion

Reply:
> Alternately, perhaps just use the ordered list in some fashion
👍

Reply:
Fixed in 972d22e.  Added a section to describe the account addresses array